server/broadcast.go

```diff
@@ -33,18 +33,33 @@ func (pi *packetInfo) broadcastMessage() {
 			pi.broadcastAll(msgs)
 		} else {
 			sendingPlayer := pi.tracker.playerByConnection(pi.conn)
+			if sendingPlayer == nil {
+				if debugMode {
+					fmt.Printf("[DirectMessage] Ignoring from %s because player no longer exists\n", getIP(pi.conn))
+				}
+				return
+			}
+
 			player := pi.tracker.playerByVerificationKey(strings.TrimLeft(vk, playerPrefix))
 			if player == nil {
+				if debugMode {
+					fmt.Printf("[DirectMessage] Ignoring to %s because player no longer exists\n", vk)
+				}
 				return
 			}
 
 			if player == sendingPlayer {
-				// Don't allow players to send messages to themselves.
+				if debugMode {
+					fmt.Printf("[DirectMessage] Ignoring to self from %s\n", sendingPlayer)
+				}
 				return
 			}
 
 			// stop sending messages after the first error
 			if err := writeMessage(player.conn, msgs); err != nil {
+				if debugMode {
+					fmt.Printf("[DirectMessage] Interrupting packet send after failed write: %s\n", err)
```

Comment:
`fmt.Printf("[DirectMessage] Error writing message to %s: %s\n", getIP(player.conn), err)`